README_WINDOWS.rst

```diff
@@ -0,0 +1,25 @@
+Borg Native on Windows
+======================
+
+Build Requirements
+------------------
+
+- VC 14.0 Compiler
+- OpenSSL Library v1.1.1c, 64bit (available at https://slproweb.com/products/Win32OpenSSL.html)
+- Patience and a lot of coffee / beer
+
+What's working
+--------------
+
+.. note::
+   The following examples assume that the `BORG_REPO` and `BORG_PASSPHRASE` environment variables are set
+   if the repo or passphrase is not explicitly given.
+
+- Borg does not crash if called with ``borg``
+- ``borg init --encryption repokey-blake2 ./demoRepo`` runs without an error/warning.
+  Note that absolute paths only work if the protocol is explicitly set to file://
+- ``borg create ::backup-{now} D:\DemoData`` works as expected.
+- ``borg list`` works as expected.
+- ``borg extract --strip-components 1 ::backup-XXXX`` works. 
+  If absolute paths are extracted, it's important to pass ``--strip-components 1`` as
+  otherwise the data is resotred to the original location!
```

Comment:
how can you extract absolute pathes?

borg only stores relative pathes into an archive, so there is no leading `/`.
if you give a leading slash at extract time, it usually does not extract anything, because it does not match.

Reply:
My test looks like this:

```
SET BORG_REPO=file://D:/borgPlayground/demoRepo
SET BORG_PASSPHRASE=test
borg create --debug --progress --list ::backup-{now} D:\\borgPlayground\demoData
REM Timestamp take from the backup created above
borg extract --debug --list --strip-components 1 ::backup-2019-08-24T11:46:17
```
If I now set a breakpoint in the `do_extract` method (https://github.com/jrast/borg/blob/win10/src/borg/archiver.py#L778) and inspect the item path, the path is absolute. In the next lines, `--strip-components` is handled and the path becomes relative. If I step through the code into `archive.extract_item` (https://github.com/jrast/borg/blob/win10/src/borg/archive.py#L685 ff), the original file would be deleted if `--strip-components` is not specified.

So at the moment it seems that on Windows absolute paths are stored in the repo!
What happens on unix systems if two directories are backed up, containing the same directory and file structure, if only relative paths are stored? On an extract of the whole archive, which file would survive?

Reply:
That's unspecified. On Linux it is assumed that you don't do stupid stuff. :-)

But I'ld consider the absolute behavior to be a bug, so either fix it within this PR or open an issue after the PR is merged.

Pathes in the archive need to be relative (check this with `borg list`).
If they are, they also must be extracted relative to the cwd.

Reply:
Ok, I tried it with borg in cygwin. An as I see it, the complete path is saved, but without a leading slash. So this is what you mean by relative? Is it always the whole path? So if I backup `~/Documents` the complete path, something like `home/jrast/Documents` will endup in the backup?

I think it's better to look at this later in a separate Issue/PR which looks at all windows related path handling stuff.

Reply:
> That's unspecified. On Linux it is assumed that you don't do stupid stuff. :-)

Maybe it would be better to prevent the user from doing stupid stuff? Even on Linux...

But after some testing with borg in cygwin, the behaviour seems to be more or less clear.
The behaviour is dependent on the path passed to `borg create`:

- If the path is relative, store the path as is
- If the path is absolute, remove the first slash to make it look relative. This is the same as if the working directory would be switched to '/'

Is this correct?

Reply:
The is no point in trying to prevent all sorts of foolish actions, it just makes the software unnecessarily complicated and won't prevent a fool from finding yet another way.

Sounds correct. Maybe you can try if it "just works" when using forward slashes on windows.

If forward slashes are supported, maybe we could just translate to them one time and be done with it.

Reply:
Forward slashes are, as far as I can tell, supported as path separator by python on windows. But absolute paths start with a drive letter, like `D:\dir\file.txt`, and there is no easy way (see below) around the drive letter (`D:`) as @Anakonda already mentioned in his inputs above.

Anyway, I don't think it makes sense to discuss this any further at the moment. Creating, listing and extracting backups is possible, which was the goal of this PR. There are still limitations for these commands, but it was never the goal to provide a fully working solution right away in a single PR. Any further functionality and improved behaviour will be implemented separate pulle requests.

It is possible to access the drive by drive label. For example, in PowerShell:
```powershell
$label = (Get-Volume | ? DriveLetter -eq "D").Path
echo $label
# \\?\Volume{972e8a8d-9362-406f-9c8f-39d1ff582a09}\
ls -l $label
# --> Prints directory listing of D:
```